hyper-capi/include/hyper.h

```diff
@@ -0,0 +1,318 @@
+#ifndef _HYPER_H
+#define _HYPER_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stddef.h>
+
+typedef struct hyper_clientconn hyper_clientconn;
+
+typedef struct hyper_clientconn_options hyper_clientconn_options;
+
+typedef struct hyper_io hyper_io;
+
+typedef struct hyper_request hyper_request;
+
+typedef struct hyper_response hyper_response;
+
+typedef struct hyper_headers hyper_headers;
+
+typedef struct hyper_body hyper_body;
+
+typedef struct hyper_buf hyper_buf;
+
+typedef struct hyper_task hyper_task;
+
+typedef struct hyper_context hyper_context;
+
+typedef struct hyper_waker hyper_waker;
+
+typedef struct hyper_executor hyper_executor;
+
+typedef enum {
+    HYPER_OK = 0,
+    HYPERE_KABOOM
+} hyper_code;
+
+typedef enum {
+	HYPER_ITER_CONTINUE,
+	HYPER_ITER_BREAK
+} hyper_iter_step;
+
+typedef enum {
+	HYPER_POLL_READY,
+	HYPER_POLL_PENDING
+} hyper_poll;
+
+// HTTP ClientConn
+
+// Starts an HTTP client connection handshake using the provided IO transport
+// and options.
+//
+// Both the `io` and the `options` are consumed in this function call.
```

Comment:
Is that also guaranteed if the function returns a `NULL`/failure? Would be good to have that clarified.